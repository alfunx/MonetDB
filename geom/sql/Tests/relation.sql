CREATE TABLE geoms (g GEOMETRY);
INSERT INTO geoms values ('POINT(10 10)');
INSERT INTO geoms values ('LINESTRING(10 10, 20 20, 30 40)');
INSERT INTO geoms values ('POLYGON((10 10, 10 20, 20 20, 20 15, 10 10))');
INSERT INTO geoms values ('POLYGON((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))');
INSERT INTO geoms values ('LINESTRING(10 10)');
INSERT INTO geoms values ('LINESTRING(10)');
INSERT INTO geoms values ('LINESTRING()');
INSERT INTO geoms values ('LINESTRING');
INSERT INTO geoms values ('');
INSERT INTO geoms values ('POINT(10)');
INSERT INTO geoms values ('POINT()');
INSERT INTO geoms values ('POINT');
INSERT INTO geoms values ('');
INSERT INTO geoms values (NULL);

SELECT * FROM geoms;

SELECT * FROM geoms where g is NOT NULL;

SELECT a.g, b.g, Equals(a.g,b.g), Disjoint(a.g,b.g), "Intersect"(a.g,b.g), Touches(a.g,b.g) FROM geoms a, geoms b where a.g is not NULL and b.g is not NULL order by a.g, b.g;

SELECT a.g, b.g, Crosses(a.g,b.g), Within(a.g,b.g), Contains(a.g,b.g), Overlaps(a.g,b.g) FROM geoms a, geoms b where a.g is not NULL and b.g is not NULL order by a.g, b.g;

SELECT a.g, b.g, Relate(a.g,b.g,'Touches') FROM geoms a, geoms b where a.g is not NULL and b.g is not NULL order by a.g, b.g;

SELECT a.g, b.g, Relate(a.g,b.g,'*F**T**F*') FROM geoms a, geoms b where a.g is not NULL and b.g is not NULL order by a.g, b.g;

DROP TABLE geoms;
