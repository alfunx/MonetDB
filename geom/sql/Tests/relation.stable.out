stdout of test 'relation` in directory 'geom/sql` itself:


# 22:42:46 >  
# 22:42:46 >  Mtimeout -timeout 180  Mserver "--config=/ufs/niels/scratch/rc/Linux-x86_64/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB4:/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB4/lib:/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB4/bin" --set "gdk_dbfarm=/ufs/niels/scratch/rc/Linux-x86_64/var/MonetDB4/dbfarm" --set "sql_logdir=/ufs/niels/scratch/rc/Linux-x86_64/var/MonetDB4/sql_logs" --set "xquery_logdir=/ufs/niels/scratch/rc/Linux-x86_64/var/MonetDB4/xquery_logs" --set mapi_open=true --set xrpc_open=true --set mapi_port=32262 --set xrpc_port=47875 --set monet_prompt= --trace "--dbname=mTests_src_sql" --dbinit="module(geom); module(sql_server);" ; echo ; echo Over..
# 22:42:46 >  

# Monet Database Server V4.18.1
# Copyright (c) 1993-2007, CWI. All rights reserved.
# Compiled for x86_64-redhat-linux-gnu/64bit with 64bit OIDs; dynamically linked.
# Visit http://monetdb.cwi.nl/ for further information.


Ready.


# 22:42:46 >  
# 22:42:46 >  Mtimeout -timeout 60 MapiClient -lsql -umonetdb -Pmonetdb --host=koala --port=32262 
# 22:42:46 >  

[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
% sys.geoms # table_name
% g # name
% geometry # type
% 0 # length
[ "POINT (10 10)"	]
[ "LINESTRING (10 10, 20 20, 30 40)"	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))"	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))"	]
[ NULL	]
% sys.geoms # table_name
% g # name
% geometry # type
% 0 # length
[ "POINT (10 10)"	]
[ "LINESTRING (10 10, 20 20, 30 40)"	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))"	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))"	]
#SELECT a.g, b.g, Equals(a.g,b.g), Disjoint(a.g,b.g), "Intersect"(a.g,b.g), Touches(a.g,b.g) FROM geoms a, geoms b where a.g is not NULL and b.g is not NULL order by a.g, b.g;
% sys.a,	sys.b,	sys.L1,	sys.L2,	sys.L3,	sys.L4 # table_name
% g,	g,	L1,	L2,	L3,	L4 # name
% geometry,	geometry,	boolean,	boolean,	boolean,	boolean # type
% 0,	0,	5,	5,	5,	5 # length
[ "POINT (10 10)",	"POINT (10 10)",	true,	false,	true,	false	]
[ "POINT (10 10)",	"LINESTRING (10 10, 20 20, 30 40)",	false,	false,	true,	true	]
[ "POINT (10 10)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false,	false,	true,	true	]
[ "POINT (10 10)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false,	false,	true,	true	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POINT (10 10)",	false,	false,	true,	true	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"LINESTRING (10 10, 20 20, 30 40)",	true,	false,	true,	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false,	false,	true,	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false,	false,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POINT (10 10)",	false,	false,	true,	true	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"LINESTRING (10 10, 20 20, 30 40)",	false,	false,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	true,	false,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false,	false,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POINT (10 10)",	false,	false,	true,	true	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"LINESTRING (10 10, 20 20, 30 40)",	false,	false,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false,	false,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	true,	false,	true,	false	]
#SELECT a.g, b.g, Crosses(a.g,b.g), Within(a.g,b.g), Contains(a.g,b.g), Overlaps(a.g,b.g) FROM geoms a, geoms b where a.g is not NULL and b.g is not NULL order by a.g, b.g;
% sys.a,	sys.b,	sys.L1,	sys.L2,	sys.L3,	sys.L4 # table_name
% g,	g,	L1,	L2,	L3,	L4 # name
% geometry,	geometry,	boolean,	boolean,	boolean,	boolean # type
% 0,	0,	5,	5,	5,	5 # length
[ "POINT (10 10)",	"POINT (10 10)",	false,	true,	true,	false	]
[ "POINT (10 10)",	"LINESTRING (10 10, 20 20, 30 40)",	false,	false,	false,	false	]
[ "POINT (10 10)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false,	false,	false,	false	]
[ "POINT (10 10)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false,	false,	false,	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POINT (10 10)",	false,	false,	false,	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"LINESTRING (10 10, 20 20, 30 40)",	false,	true,	true,	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	true,	false,	false,	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	true,	false,	false,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POINT (10 10)",	false,	false,	false,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"LINESTRING (10 10, 20 20, 30 40)",	true,	false,	false,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false,	true,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false,	false,	true,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POINT (10 10)",	false,	false,	false,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"LINESTRING (10 10, 20 20, 30 40)",	true,	false,	false,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false,	true,	false,	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false,	true,	true,	false	]
#SELECT a.g, b.g, Relate(a.g,b.g,'*F**T**F*') FROM geoms a, geoms b where a.g is not NULL and b.g is not NULL order by a.g, b.g;
% sys.a,	sys.b,	sys.L1 # table_name
% g,	g,	L1 # name
% geometry,	geometry,	boolean # type
% 0,	0,	5 # length
[ "POINT (10 10)",	"POINT (10 10)",	false	]
[ "POINT (10 10)",	"LINESTRING (10 10, 20 20, 30 40)",	false	]
[ "POINT (10 10)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false	]
[ "POINT (10 10)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POINT (10 10)",	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"LINESTRING (10 10, 20 20, 30 40)",	true	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	false	]
[ "LINESTRING (10 10, 20 20, 30 40)",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POINT (10 10)",	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"LINESTRING (10 10, 20 20, 30 40)",	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	true	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POINT (10 10)",	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"LINESTRING (10 10, 20 20, 30 40)",	false	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10))",	true	]
[ "POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	"POLYGON ((10 10, 10 20, 20 20, 20 15, 10 10), (15 15, 15 20, 10 15, 15 15))",	true	]

# 22:42:47 >  
# 22:42:47 >  Done.
# 22:42:47 >  

